desc/builder/enum.go

```diff
@@ -250,20 +250,17 @@ func (eb *EnumBuilder) buildProto(path []int32, sourceInfo *descriptorpb.SourceC
 
 	if len(needNumbersAssigned) > 0 {
 		tags := make([]int, len(values)-len(needNumbersAssigned))
-		for i, ev := range values {
+		tagsIndex := 0
+		for _, ev := range values {
 			tag := ev.GetNumber()
 			if tag != 0 {
-				tags[i] = int(tag)
+				tags[tagsIndex] = int(tag)
+				tagsIndex++
 			}
 		}
 		sort.Ints(tags)
 		t := 0
-		ti := sort.Search(len(tags), func(i int) bool {
-			return tags[i] >= 0
-		})
-		if ti < len(tags) {
-			tags = tags[ti:]
-		}
```

Comment:
I don't think it's correct to remove this. If a value is explicitly set to a negative value, then the logic below will break. For example, if you set explicitly set tags -2, -1, and 0, but leave three values without a number, then below in the first iteration it will see `t == tags[0]` is false and incorrectly think that zero is available for use.

Reply:
yes, it makes sense, thank you.

I put this condition back in place.